image_data_sar/review.py

```diff
@@ -0,0 +1,201 @@
+            and (np.abs(dq3.yaw * 3600) < 1)
+            and np.allclose([att2.roll, att3.roll], att1.roll, atol=0.05, rtol=0))
+
+
+def pok(ok):
+    return 'OK' if ok else 'Not OK'
+
+
+def main():
+    opt = get_options()
+    sc = custom_read_starcheck(opt.starcheck)
+
+    acas = pickle.load(gzip.open(opt.pkl))
+    aca_arr = sorted(acas.values(), key=lambda aca: aca.date)
+    obsids = np.array([aca.obsid for aca in aca_arr])
+
+    er_cands = [aca.obsid for aca in aca_arr
+                if aca.obsid > 39000 and np.abs(aca.call_args['dither_guide'][0] - 16) < .1]
+
+    for obsid in er_cands:
+
+        idx = np.flatnonzero(obsids == obsid)[0]
+        obs1 = obsid
+        obs2 = obsids[idx + 1]
+        obs3 = obsids[idx + 2]
+        if idx + 3 == len(obsids):
+            if opt.sched_stop is None:
+                raise ValueError(f"Need --sched-stop to check duration of {obs3}")
+            else:
+                aca_arr.append({'sched_stop': CxoTime(opt.sched_stop)})
+
+        cat1 = starcheck_for_obsid(obs1, sc)
+        cat2 = starcheck_for_obsid(obs2, sc)
+        cat3 = starcheck_for_obsid(obs3, sc)
+
+        check1 = ((not has_sc_critical(cat1['orig_text']))
+                  and (not has_sc_critical(cat2['orig_text'], ok6x6=True))
+                  and (not has_sc_critical(cat3['orig_text']))
+                  and (not has_proseco_critical(acas[obs1]))
+                  and (not has_proseco_critical(acas[obs2]))
+                  and (not has_proseco_critical(acas[obs3])))
+
+        print(f"For SAR activity during obsids {obs1, obs2, obs3}")
+        print(f"[{pok(check1)}] - Each obs meets normal ACA review requirements (except 6x6 in ER)")
+
+        check2 = obs1_setup_ok(cat1)
+
+        print(f"[{pok(check2)}] - First obs has 16x16 dither with period=2000.0, 1414.8, ",
+              "and image size=8x8 pixels")
+
+        check3 = obs2_setup_ok(cat2)
+
+        print(f"[{pok(check3)}] - Second obs has 8x8 dither with period=1000.0, 707.1, ",
+              "and image size=6x6 pixels")
+
+        check4 = obs3_setup_ok(cat3)
+
+        print(f"[{pok(check4)}] - Third obs has 8x8 dither with period=1000.0, 707.1, ",
+              "and image size=8x8 pixels")
+
+        durations = get_durations(aca_arr)
+        check5 = (check_duration(obs1, obsids, durations)
+                  and check_duration(obs2, obsids, durations)
+                  and check_duration(obs3, obsids, durations))
+
+        print(f"[{pok(check5)}] - Each obs at least 4 ks")
+
+        check6 = same_guide_stars(obs1, obs2, obs3, acas)
+        print(f"[{pok(check6)}] - All guide stars are the same")
+
+        check7 = same_attitude(obs1, obs2, obs3, acas)
+        print(f"[{pok(check7)}] - All obs are at the same attitude")
+
+        print("")
+        print("Compare included stars to previously provided list by hand:")
+        print(f"include_ids_guide stars are {acas[obs1].call_args.get('include_ids_guide')}")
```

Comment:
This was the only line not so useful. It wrote this:
```
include_ids_guide stars are [6.51566200e+08 6.51562736e+08 6.51559472e+08 6.51564312e+08
 6.51560072e+08 6.51172128e+08 6.51558984e+08 6.51559968e+08]
```
which I think in some cases might be missing one digit. Still ok, but if this is a numpy array, perhaps one can change the type to int.

Reply:
Thanks!  I just updated to int, though I didn't see this "print in scientific notation" issue in my test environment so didn't really confirm the change made the problem go away.